// ng2-tweet/lib/components/ng2-tweet/ng2-tweet.component.js
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
var core_1 = require('@angular/core');
var index_1 = require('../../services/index');
var Ng2TweetComponent = (function () {
    function Ng2TweetComponent(element, ng2TweetService) {
        this.element = element;
        this.ng2TweetService = ng2TweetService;
    }
    Ng2TweetComponent.prototype.ngOnInit = function () {
    };
    Ng2TweetComponent.prototype.ngAfterViewInit = function () {
        var _this = this;
        //MAKE SURE TWITTER WIDGET SCRIPT IS LOADED IN HEAD...
        this.ng2TweetService.LoadScript().subscribe(
        //SUCCESS, WE HAVE TWITTER WIDGETS JS FILE LOADED...
        function (twttr) {
            var nativeElement = _this.element.nativeElement;
            window['twttr'].widgets.createTweet(_this.tweetId, nativeElement, {}).then(function success(embed) {
                //console.log('Created tweet widget: ', embed);
            }).catch(function creationError(message) {
                //console.log('Could not create widget: ', message);
            });
        }, 
        //ERROR
        function (err) {
            console.log('****  ERROR LOADING TWITTER WIDGET', err);
        }, 
        //COMPLETE
        function () {
        });
    };
    Ng2TweetComponent.prototype.onTwitterLoaded = function (twttr) {
        console.log('TWITTER LOADED YO', twttr);
    };
    ;
    __decorate([
        core_1.Input(), 
        __metadata('design:type', String)
    ], Ng2TweetComponent.prototype, "tweetId", void 0);
    Ng2TweetComponent = __decorate([
        core_1.Component({
            selector: 'ng2-tweet',
            template: "",
            styles: [""],
            providers: []
        }), 
        __metadata('design:paramtypes', [core_1.ElementRef, index_1.Ng2TweetService])
    ], Ng2TweetComponent);
    return Ng2TweetComponent;
}());
exports.Ng2TweetComponent = Ng2TweetComponent;
